### PR TITLE
update Guava

### PR DESCRIPTION
**Pull request by elharo**
@loosebazooka

### CODE REVIEW

**Review by elharo — COMMENTED** [review]
go/friendly-ping

**Comment by loosebazooka** [comment]
I'm not actually sure this is the right guava version, the code is a little old, but I think we need to use the jre guava as this is usually included in the app-engine component. You can try running through an example to verify this.

**Comment by elharo** [comment]
Can you elaborate? How do the classpaths intersect?

**Comment by loosebazooka** [comment]
There are some docs here: https://cloud.google.com/endpoints/docs/frameworks/java/migrating-android

**Comment by elharo** [comment]
This is a dependency of the maven plugin, I think. It shouldn't be a dependency of the artifacts that are built, should it? That said, we can migrate this to the jre version if you prefer.

**Comment by elharo** [comment]
weird, switchign to jre version might have broken the build. Or it might be flaky. retrying to see.

**Comment by loosebazooka** [comment]
I don't think switching back and forth between android and jre is an effective way to understand the issue, so I looked into this.  The error has nothing to do with the code in this project, rather it fails at help info generation during the execution of the `org.apache.maven.plugins:maven-plugin-plugin:3.3:helpmojo`. More info on the incompatibility with the maven-plugin-plugin and guava: https://stackoverflow.com/questions/48547145/guava-breaks-maven-plugin

You should probably update the maven-plugin-plugin to latest: https://search.maven.org/artifact/org.apache.maven.plugins/maven-plugin-plugin/3.6.0/maven-plugin

Maybe even update the repositories maven wrapper while you're here:
https://github.com/takari/maven-wrapper (there seems to be an effort to migrate this to maven core, but this might still be enough at the moment to upgrade to latest maven: 3.6.3?)

File: pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <version>21.0</version>
+      <version>30.0-android</version>
```
Comment (loosebazooka) [inline pom.xml:74]: with the updated maven plugin plugin, this should be jre.

**Review by elharo — COMMENTED** [review]
Ping. All checks are passing

**Review by elharo — COMMENTED** [review]
ping ping ping
Github just alerted me about this repo again

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**